napari/_qt/widgets/qt_layerlist.py

```diff
@@ -97,12 +98,32 @@ def __init__(self, layers):
         self.layers.events.inserted.connect(self._add)
         self.layers.events.removed.connect(self._remove)
         self.layers.events.reordered.connect(self._reorder)
+        self.layers.selection.events.connect(self._on_selection)
 
         self._drag_start_position = np.zeros(2)
         self._drag_name = None
 
         self.chunk_receiver = _create_chunk_receiver(self)
 
+    def _on_selection(self, event):
+        if event.type == 'added':
+            for layer in event.value:
+                w = self._find_widget(layer)
+                if w:
+                    w.setSelected(True)
+            self._ensure_visible(list(event.value)[-1])
+        elif event.type == 'removed':
+            for layer in event.value:
+                w = self._find_widget(layer)
+                if w:
+                    w.setSelected(False)
+
+    def _find_widget(self, layer):
+        for i in range(self.vbox_layout.count()):
+            w = self.vbox_layout.itemAt(i).widget()
+            if getattr(w, 'layer', None) == layer:
+                return w
```

Comment:
How many layers can a user have ? Those two things smell quadratic to me on a "(un)select all layers" action (assuming `self.vbox_layout.itemAt(i)` is constant time); I doubt it will have much impact so that might be ok.

Reply:
yeah, and this will all be gone in basically the next pr.  (the whole qt_layerlist file is going... this is just an intermediate PR to have it in the commit history)

Reply:
Ah, all good then !